pkg/forwarder/backoff.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package forwarder
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	// This controls the overlap between consecutive retry interval ranges. When
+	// set to `2`, there is a guarantee that there will be no overlap. The overlap
+	// will asymptotically approach 50% the higher the value is set.
+	minBackoffFactor = 2
+
+	// This controls the rate of exponential growth. Also, you can calculate the start
+	// of the very first retry interval range by evaluating the following expression:
+	// baseBackoffTime / minBackoffFactor * 2
```

Comment:
It's actually random in `[baseBackoffTime / minBackoffFactor * 2; baseBackoffTime * 2[` no ? (when `numErrors == 1`)

Reply:
No actually you can determine the start of the first range using that `baseBackoffTime / minBackoffFactor * 2`

Reply:
ah right, didn't see the `start` keyword